Resources/views/FragmentAdmin/form_list_script.html.twig

```diff
@@ -0,0 +1,78 @@
+{% endif %}
+
+<script type="text/javascript">
+    // setup an "add a text" link
+    var $addTextLink_{{ currentId }} = $('<a href="#" class="btn btn-success btn-sm sonata-ba-action"><i class="fa fa-plus-circle"></i>&nbsp;{{ btn_plus_title|trans({}, 'SonataArticleBundle') }}</a>');
+
+    function init_{{ currentId }}() {
+        // Get the ul that holds the collection of texts
+        var $collectionHolder = $('#{{ currentId }}');
+
+        // add the "Ajouter un texte" anchor
+        $collectionHolder.append($addTextLink_{{ currentId }});
+
+        // add the remove anchor
+        $collectionHolder.find('.sonata-collection-row-without-label').each(function(index) {
+            addDeleteLink($(this));
+        });
+
+        // count the current form inputs we have (e.g. 2), use that as the new
+        // index when inserting a new item (e.g. 2)
+        $collectionHolder.data('index', $collectionHolder.find(':input').length);
+
+        $addTextLink_{{ currentId }}.on('click', function(e) {
+            // prevent the link from creating a "#" on the URL
+            e.preventDefault();
+
+            // add a new text form (see code block below)
+            addTextForm_{{ currentId }}($collectionHolder, $addTextLink_{{ currentId }});
+        });
+    }
+
+    function addDeleteLink($formPart) {
+        var $removeForm = $('<button type="button" class="btn btn-danger btn-sm sonata-ba-action pull-right remove-service-block-button">&nbsp;{{ 'delete'|trans({}, 'SonataArticleBundle') }}</button><br/>');
+        $removeForm.insertBefore($formPart);
+
+        $removeForm.on('click', function(e) {
+            // prevent the link from creating a "#" on the URL
+            e.preventDefault();
+
+            // remove the part of form
+            $formPart.remove();
+            $(this).remove();
+        });
+    }
+
+    function addTextForm_{{ currentId }}($collectionHolder, $addTextLink) {
+        // Get the data-prototype explained earlier
+        var prototype = $collectionHolder.data('prototype');
+
+        // get the new index
+        var index = $collectionHolder.data('index');
+
+        // Replace '$$name$$' in the prototype's HTML to
+        // instead be a number based on how many items we have
+        var newForm = prototype.replace(/__name__/g, index);
+        newForm = '<button type="button" class="close"><span aria-hidden="true">&times;</span><span class="sr-only">{{ 'delete'|trans({}, 'SonataArticleBundle') }}</span></button><br/>' + newForm;
+
+        // increase the index with one for the next item
+        $collectionHolder.data('index', index + 1);
+
+        var $newFormDiv = $('<div></div>').append(newForm);
+        $addTextLink.before($newFormDiv);
+
+        $('.close').on('click', function(e) {
+            e.preventDefault();
+
+            $(this).parent().remove();
+
+            return false;
+        });
+    }
+
+    init_{{ currentId}}();
+</script>
```

Comment:
Also, creating a javascript function with an id in its name looks really weird :/ Why do you need that?